File: backend-springboot/src/main/java/com/tfg/backend/SignalEnvelope/dto/SignalMessageWSDto.java
package com.tfg.backend.SignalEnvelope.dto;

import java.time.LocalDateTime;

public class SignalMessageWSDto {
    private final Long envelopeId;
    private final Long senderUserId;
    private final Long conversationId;
    private final String conversationType;
    private final Short cypherTextType;
    private final String cypherTextB64;
    private final LocalDateTime createdAt;

    public SignalMessageWSDto(
        Long envelopeId,
        Long senderUserId,
        Long conversationId,
        String conversationType,
        Short cypherTextType,
        String cypherTextB64,
        LocalDateTime createdAt
    ) {
        this.envelopeId = envelopeId;
        this.senderUserId = senderUserId;
        this.conversationId = conversationId;
        this.conversationType = conversationType;
        this.cypherTextType = cypherTextType;
        this.cypherTextB64 = cypherTextB64;
        this.createdAt = createdAt;
    }

    public Long getEnvelopeId() {
        return envelopeId;
    }

    public Long getSenderUserId() {
        return senderUserId;
    }

    public Long getConversationId() {
        return conversationId;
    }

    public String getConversationType() {
        return conversationType;
    }

    public Short getCypherTextType() {
        return cypherTextType;
    }

    public String getCypherTextB64() {
        return cypherTextB64;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }
}

File: mobile-app/app/src/main/java/com/example/mobile_app/data/model/signal/SignalMessageWSDto.kt
package com.example.mobile_app.data.model.signal

/**
 * DTO para mensajes Signal recibidos via WebSocket.
 *
 * `createdAt` se modela como String: el backend serializa LocalDateTime como
 * cadena ISO-8601 ("2026-05-17T18:14:42.401976378") y Moshi no trae adapter
 * para java.time.LocalDateTime.
 */
data class SignalMessageWSDto(
    val envelopeId: Long,
    val senderUserId: Long,
    val conversationId: Long,
    val conversationType: String,
    val cypherTextType: Short,
    val cypherTextB64: String,
    val createdAt: String? = null
)
